src/plugins/index_pattern_field_editor/public/components/field_format_editor/format_editor.tsx

```diff
@@ -6,43 +6,46 @@
  * Side Public License, v 1.
  */
 
-import React, { PureComponent, Fragment } from 'react';
-import { FieldFormat } from 'src/plugins/data/public';
+import { EuiDelayRender, EuiLoadingContent } from '@elastic/eui';
+import { memoize } from 'lodash';
+import React, { PureComponent, LazyExoticComponent } from 'react';
+import type { FieldFormat } from 'src/plugins/data/public';
+import { FieldFormatEditorFactory, FieldFormatEditor } from './editors';
 
 export interface FormatEditorProps {
   fieldType: string;
   fieldFormat: FieldFormat;
   fieldFormatId: string;
   fieldFormatParams: { [key: string]: unknown };
   fieldFormatEditors: any;
-  onChange: (change: { fieldType: string; [key: string]: any }) => void;
```

Comment:
After improving types, typescript highlighted type mismatch here. Looks like there is no `fieldType: string;` emitted from the editors so I removed it from here to fix types